ContactManager/ContactManager/ContactManager.cs

```diff
@@ -0,0 +1,419 @@
+        /// </summary>
+        /// <param name="menuTitle">Title of the Menu.</param>
+        /// <param name="menuActions">A dictionary mapping menu choices to their corresponding actions.</param>
+        /// 
+        void HandleMenuActions(string menuTitle, Dictionary<int, (string, Action action)> menuActions, bool clearConsole = true) {
+            do {
+                if (clearConsole)
+                    Console.Clear();
+                DisplayMenu(menuTitle, menuActions);
+                Console.Write("\n[Menu] Enter your choice: ");
+                string input = Console.ReadLine();
+                bool isNumber = int.TryParse(input, out int choice);
+
+                if (isNumber && menuActions.ContainsKey(choice) && choice != 4) {
+                    menuActions[choice].action.Invoke();
+                } else if (isNumber && choice == menuActions.Count) {
+                    return;
+                } else {
+                    Console.WriteLine("[Error] Invalid choice!");
+                    PromptForContinuation();
+                }
+
+            } while (!IsOperationSuccessful);
+        }
+
+        /// <summary>
+        /// Retrieves a contact's details by searching for it based on a specified attribute.
+        /// </summary>
+        /// <param name="ContactBy">The attribute to search the contact by (e.g., "Name", "Phone Number", "Email").</param>
+        /// <returns>
+        /// The <see cref="ContactDetails"/> object representing the contact if found; otherwise, null.
+        /// </returns>
+        ContactDetails GetContactDetailByAttribute(string ContactBy) {
+            string input = GetContactInformation(ContactBy);
+            ContactDetails contact = FindContactByAttribute(ContactBy, input);
+            return contact;
+        }
+
+        /// <summary>
+        /// Displays a submenu for editing specific fields of a contact and performs the selected action.
+        /// </summary>
+        /// <param name="ContactToEdit">The <see cref="ContactDetails"/> object representing the contact to be edited.</param>
+        void EditBy(ContactDetails ContactToEdit) {
+            var EditByMenuActions = new Dictionary<int, (string, Action action)> {
+                { 1, ("Edit Name", () => UpdateContactField(_consoleName, true, value => ContactToEdit.Name = value))},
+                { 2, ("Edit Phone Number", () => UpdateContactField(_consolePhoneNumber, true, value => ContactToEdit.PhoneNumber = value))},
+                { 3, ("Edit Email", () => UpdateContactField(_consoleEmail, true, value => ContactToEdit.Email = value))},
+                { 4, ("Edit Additional Information", () => UpdateContactField(_consoleAdditionalInformation, false, value => ContactToEdit.AdditionalInformation = value))},
+                { 5, ("Main Menu", () => { IsOperationSuccessful = true; })}
+            };
+
+            bool isValidChoice = false;
```

Comment:
I don't see this boolean used anywhere? Is it needed

```suggestion
            bool isValidChoice = false;
```